packages/content-management/favorites/favorites_public/src/favorites_client.ts

```diff
@@ -9,12 +9,16 @@
 
 import type { HttpStart } from '@kbn/core-http-browser';
 import type { UsageCollectionStart } from '@kbn/usage-collection-plugin/public';
-import type { GetFavoritesResponse } from '@kbn/content-management-favorites-server';
+import type {
+  GetFavoritesResponse,
+  AddFavoriteResponse,
+  RemoveFavoriteResponse,
+} from '@kbn/content-management-favorites-server';
 
 export interface FavoritesClientPublic {
   getFavorites(): Promise<GetFavoritesResponse>;
-  addFavorite({ id }: { id: string }): Promise<GetFavoritesResponse>;
-  removeFavorite({ id }: { id: string }): Promise<GetFavoritesResponse>;
+  addFavorite({ id, metadata }: { id: string; metadata?: object }): Promise<AddFavoriteResponse>;
```

Comment:
TODO: type it as generic so make usage nicer ? Also depending on how esql discover will be using it, might need to update other client-side things like hooks and components